packages/core/src/http/HttpAgentImpl.ts

```diff
@@ -416,24 +416,17 @@ export class HttpAgentImpl extends HttpAgent {
    *         internally.
    */
   _prepareOptions(
-    options?: Partial<HttpAgentRequestOptions>
+    options: Partial<HttpAgentRequestOptions> = {}
   ): HttpAgentRequestOptions {
-    const composedOptions = {
-      ...this._defaultRequestOptions,
-      ...options,
-    };
-    let cookieHeader = {};
+    const composedOptions = mergeDeep(this._defaultRequestOptions, options);
 
-    if (composedOptions.withCredentials) {
-      cookieHeader = { Cookie: this._cookie.getCookiesStringForCookieHeader() };
+    if (composedOptions.fetchOptions?.credentials === 'include') {
+      // mock default browser behavior for server-side (sending auth cookie)
```

Comment:
Not sure I understand what this comment mean, can you elaborate?

Reply:
According to @mjancarik this part makes sure that the auth cookie is set even when fetching server-side... (e.g. in node), since there's no browser to do it automatically. Not sure how better to explain?